opamp/observiq/observiq_client_test.go

```diff
@@ -251,7 +257,7 @@ func TestClientConnect(t *testing.T) {
 				mockOpAmpClient.On("Start", mock.Anything, mock.Anything).Return(nil).Run(func(args mock.Arguments) {
 					settings := args.Get(1).(types.StartSettings)
 					assert.Equal(t, expectedSettings.OpAMPServerURL, settings.OpAMPServerURL)
-					assert.Equal(t, expectedSettings.AuthorizationHeader, settings.AuthorizationHeader)
+					assert.True(t, reflect.DeepEqual(expectedSettings.Header, settings.Header), "Header elements should match")
```

Comment:
This can just be `assert.Equal` as it does a [reflect.DeepEqual](https://github.com/stretchr/testify/blob/083ff1c0449867d0d8d456483ee5fab8e0c0e1e6/assert/assertions.go#L58-L76)